clio-app/Resource/Modifiers/HelpPopup.swift

```diff
@@ -0,0 +1,127 @@
+            .overlay {
+                GeometryReader { geo in
+                    ZStack {
+                        if helpAlert {
+                            Color.black.opacity(0.3)
+                                .ignoresSafeArea()
+                                .transition(.opacity)
+                                .onTapGesture {
+                                    changeVisibility()
+                                }
+                            
+                             CustomAlert(
+                                isPopupPresented: $helpAlert,
+                                title: title,
+                                text: text
+                            )
+                            .ignoresSafeArea()
+                            .padding()
+                            .transition(.scale.combined(with: .offset(
+                                x: geo.size.width,
+                                y: -geo.size.height
+                            )))
+                            .clipped()
+                            .scaleEffect(1)
+                            .offset(
+                                x: helpAlert ? 0 : -2000,
+                                y: helpAlert ? 0 : -2000
+                            )
+                        }
+                    }
+                }
+            }
+            .toolbar {
+                ToolbarItem(placement: .navigationBarTrailing) {
+                    Button() {
+                        UIImpactFeedbackGenerator(style: .soft).impactOccurred()
+                        changeVisibility()
+                    }label: {
+                        Image(systemName: "exclamationmark.circle.fill")
+                            .resizable()
+                            .scaledToFill()
+                            .foregroundColor(.lapisLazuli)
+                            .frame(width: 38, height: 38)
+                            .background {
+                                Color.white
+                                    .clipShape(Circle())
+                                    .shadow(radius: 2)
+                            }
+                            .padding(.top, 2)
+                            .padding(.bottom,5)
+                            .padding(.leading, 10)
+                        
+                    }
+                    .transition(.move(edge: .bottom).combined(with: .move(edge: .leading)))
+                    .disabled(helpAlert)
+                    .opacity(helpAlert ? 0 : 1)
+                }
+            }
+    }
+    
+    func changeVisibility() {
+        withAnimation(.easeInOut(duration: 0.3)) {
+            helpAlert.toggle()
+        }
+    }
+    
+}
+
+
+extension HelpArea {
+    enum HelpType {
+        case start
+        case addingPlayer
+        case selectTheme
+        case selectPlayer
+        case capturePhoto
+        case captureDescription
+        case presentResult
+        case readyForResults
```

Comment:
Essa implementação de uma descrição por View é uma boa escolha. A única coisa que eu acredito que vá ser um problema no futuro é quando a gente for adicionando novas views. É possível que a gente esqueça de adicionar um novo case no enum e acabe não percebendo.

Pra evitar isso, eu acredito que a melhor escolha seria reutilizar o enum de Views que está definido lá no `Router`. Praticamente todos os cases que estão nesse enum tão definidos lá:

```swift
enum Views: Hashable {
    case Start
    case AddPlayers
    case RaffleTheme
    case SelectPlayer
    case PhotoArtifact
    case DescriptionArtifact
    case ResultsVisualization
}
```
Isso iria auxiliar a gente em quando uma nova view for adicionada ao Router o compilador do swift alertar que o switch do `HelpType` não está cobrindo todos os casos.

A única diferença de implementação foi que no case de Help possui um case para `presentResult` e outro para `readyForResults`. Essa diferença é na verdade um débito técnico que a gente deixou na implementação da tela de resultados ao não utilizar o Router e sim o navigationLink. Eu vou iniciar a implementação dessa parte da tela de resultados e enviar para a `dev`. Assim, os dois enums vão ficar iguais.

Reply:
Acabei de atualizar a dev com essas mudanças!